src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java

```diff
@@ -0,0 +1,224 @@
+
+  private static void parseJsonData(String data, Map<String, JsonObject> dataMap) {
+    String[] lines = data.split("\n");
+    for (String line : lines) {
+      JsonObject json = gson.fromJson(line, JsonObject.class);
+      String id = json.get("id").getAsString();
+      dataMap.put(id, json);
+    }
+  }
+
+  private static void parseCsvData(String data, Map<String, JsonObject> dataMap) {
+    String[] lines = data.split("\n");
+    String[] headers = lines[0].split(",");
+
+    for (int i = 1; i < lines.length; i++) {
+      String[] values = lines[i].split(",");
+      JsonObject jsonObject = new JsonObject();
+      for (int j = 0; j < headers.length; j++) {
+        jsonObject.addProperty(headers[j], values[j]);
+      }
+      String id = values[0];
+      dataMap.put(id, jsonObject);
+    }
+  }
+
+  /**
+   * Converts data from a CSV file to a map of JSON objects.
+   *
+   * @return A map with identifiers (e.g., ID from the first column) as keys and JSON objects as values.
+   * @throws IOException If there's an error reading the CSV file.
+   */
+  public static Map<String, JsonObject> readCsvAndConvertToJson() {
+    Map<String, JsonObject> csvDataMap = new HashMap<>();
+    try (CSVReader csvReader = new CSVReader(new java.io.FileReader(csvFilePath))) {
+      // Read the header line to get column names
+      String[] headers = csvReader.readNext();
+
+      String[] line;
+      while ((line = csvReader.readNext()) != null) {
+        JsonObject jsonObject = new JsonObject();
+
+        for (int j = 0; j < headers.length; j++) {
+          jsonObject.addProperty(headers[j], line[j]);
+        }
+        String id = line[0];
+        csvDataMap.put(id, jsonObject);
+      }
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    return csvDataMap;
+  }
+
+  /**
+   * Converts Avro files to JSON objects with keys and stores them in a map.
+   *
+   * @return A map of keys to JSON objects representing the Avro data.
+   * @throws IOException If an IO error occurs during Avro to JSON conversion.
+   */
+  public static Map<String, JsonObject> convertAvroToJsonWithKeys() throws IOException {
+    File avroFile = new File(avroFilePath);
+    DatumReader<GenericRecord> datumReader = new GenericDatumReader<>();
+    Map<String, JsonObject> avroDataMap = new HashMap<>();
+    try (FileReader<GenericRecord> dataFileReader = DataFileReader.openReader(avroFile, datumReader)) {
+      int keyCounter = 1;
+      while (dataFileReader.hasNext()) {
+        GenericRecord record = dataFileReader.next();
+        String json = record.toString();
+        JsonObject jsonObject = gson.fromJson(json, JsonObject.class);
+        avroDataMap.put(String.valueOf(keyCounter), jsonObject);
+        keyCounter++;
+      }
+    }
+    return avroDataMap;
+  }
+
+  /**
+   * Enumeration representing different data file formats that can be used
+   * for storing and retrieving data, such as JSON and CSV.
+   */
+  public enum DataFormat {
```

Comment:
Do we need an enum for only 2 values ?

This is definitely a good practice and if you wanna use it I believe you can create it separately in the package.

Reply:
Created a separate enum in a util folder for these two different file formats.